packages/server/src/rooms.ts

```diff
@@ -64,3 +64,30 @@ async function doJoinRoom(io: IOServer, socket: Socket, store: Store, room: Room
     io.in(room.name).emit(GameEvent.CurrentState, state);
   }
 }
+
+export function updateGameState(io: IOServer, store: Store, roomName: string, gameState: GameState) {
```

Comment:
This file is supposed to be about room management, not game state...

Reply:
so should I create a games file? The problem is that game state right now it's coupled with room, and we are creating the game state inside the room already that's why I added it here, maybe I move that part as well

Reply:
ok that part is in utils actually, so next question is, is there a difference between this updateGameState and generateGameState?

The difference is that this function emit some events to the socket while the other is not. Should we move all the thing that emits to a common folder maybe?

Reply:
`generateGameState` is kinda temporary. If the game is now playable, it should probably live in a game-related module.

In terms of separation... Yeah, we might need some refactoring. I guess `Game` is anything game logic related, and Room is anything around joining/players etc. I say we merge what we have, and review/refactor later.